rfcs/034-location_location_location/README.md

```diff
@@ -0,0 +1,133 @@
+Please check if the information contained is accurate and complete.
+
+
+
+## The existing approach
+
+The Catalogue API contains *Works*, and Works contain *Items*.
+In turn, each Item can have *Locations*, which tell the user how they can get the item.
+We're not planning to change the Item model; we want to consider if the Location model is fit for purpose.
+
+<details>
+<summary>Examples of physical and digital locations in the existing model</summary>
+
+Physical location:
+
+```json
+{
+  "locationType": {
+    "id": "scmac",
+    "label": "Closed stores Arch. & MSS",
+    "type": "LocationType"
+  },
+  "label": "Closed stores Arch. & MSS",
+  "accessConditions": [
+    {
+      "status": {
+        "id": "open",
+        "label": "Open",
+        "type": "AccessStatus"
+      },
+      "terms": "The papers are available subject to the usual conditions of access to Archives and Manuscripts material.",
+      "type": "AccessCondition"
+    }
+  ],
+  "type": "PhysicalLocation"
+}
+```
+
+  Digital location:
+
+```json
+{
+  "locationType": {
+    "id": "iiif-image",
+    "label": "IIIF Image API",
+    "type": "LocationType"
+  },
+  "url": "https://iiif.wellcomecollection.org/image/A0000681.jpg/info.json",
+  "credit": "Royal Veterinary College",
+  "license": {
+    "id": "cc-by-nc",
+    "label": "Attribution-NonCommercial 4.0 International (CC BY-NC 4.0)",
+    "url": "https://creativecommons.org/licenses/by-nc/4.0/",
+    "type": "License"
+  },
+  "accessConditions": [],
+  "type": "DigitalLocation"
+}
+```
+
+</details>
+
+
+
+## What should a location tell you?
+
+When somebody looks at an item, they want to know two things about the location:
+
+*   **Am I allowed to see it?**
+    This is encoded by the *access conditions* field, which includes an *access status* (drawn from a fixed vocabulary: open, closed, requires-permission, etc.) and an *access description* (free-text, written by cataloguers).
+
+*   **If I'm allowed to see it, where can I find it?**
+    For example: where it sits on the physical shelves within the library, or a particular URL, or a IIIF resource.
+
+We need to support filtering and aggregating by location.
+For example, users should be able to find all the material that's openly available online (if they're working remotely), or all the material in the library (if they're in the building).
+
+
+
+## What locations do we need to support?
+
+These are the types of location our model needs to support:
+
+*   **Open shelves.**
+
+    This should include a *shelf location* (e.g. History of Medicine Collection) and a *shelfmark* (e.g. TW.AL) so that a user can find it.
+
+*   **Closed stores.**
+
+    Although the MARC records include a shelf location and a shelfmark, the Collections team would prefer this not to be exposed publicly.
+
+    Do we need to expose this information for the requesting service?
+
+    -   Can something be requested faster if it's stored on-site?
```

Comment:
Follow-up question (also in the body of the RFC):

> We don't need to expose the full detail of our closed stores, but do we want to expose the difference between onsite/offsite? The latter will take longer to retrieve, and may require a manual request to Library Enquiries. Is this something that should be exposed by the Catalogue API, or only by the Requesting service?

Reply:
Good question. Requesting should not be possible in the first instance for these items, but we may add it later as a manual request rather than a hold in Sierra. Whether it should be here is more to do with how likely it is to change and where exactly it sits in the model alongside other access methods.